### PR TITLE
Disable MariaDB persistence to avoid leaking PVCs

### PR DESCRIPTION
**Pull request by nan-yu**
Mariadb persistent storage is enabled by default, which creates a PVC in the namespace. When wordpress is deleted, the PVC is left behind after the test.

This commit updates the chart values to disable mariadb persistent storage.

### CODE REVIEW

File: e2e/testcases/helm_sync_test.go
Comment (karlkfi) [inline e2e/testcases/helm_sync_test.go:341]: Is this created indirectly? Why doesn't the normal cleanup delete it? Are we missing Cleanup that removes the helm chart and deletes all the bits in the inventory?

Comment (nan-yu) [reply e2e/testcases/helm_sync_test.go:341]: It is not an object declared in the chart and managed by Config Sync. I believe MariaDB creates it on the cluster to persist data, but never deletes it even when MariaDB is deleted.

File: e2e/testcases/helm_sync_test.go
Comment (sdowell) [inline e2e/testcases/helm_sync_test.go:106]: Is there a `PVC` created in the default namespace as well by the updated RootSync on line `102`?

File: e2e/testcases/helm_sync_test.go
Comment (sdowell) [inline e2e/testcases/helm_sync_test.go:103]: Is this `Cleanup` associated with the change in namespace on the line above? From reading this it sounds like it's associated with the initial apply on line `61`. In which case, I'm worried we won't reach this `Cleanup` block before the validations fail. Am I missing something? If not, can this `Cleanup` be moved up to around line `62` to ensure it runs?

File: e2e/testcases/helm_sync_test.go
Comment (nan-yu) [reply e2e/testcases/helm_sync_test.go:106]: Yes, and the `deleteWordpressPVC` function deletes the PVC in the `default` namespace. There is also a PVC created in the `wordpress` namespace from `root-sync-helm-chart-cr.yaml` that is applied in line 61. That is deleted as a side-effect of pruning the `wordpress` namespace.

File: e2e/testcases/helm_sync_test.go
Comment (karlkfi) [reply e2e/testcases/helm_sync_test.go:341]: Is this a generic problem users will have? Do we need to document it?

How is the PVC created? Is it the result of a Deployment field?

Is there a way to configure the chart not to use persistent storage?

File: e2e/testcases/helm_sync_test.go
Comment (sdowell) [reply e2e/testcases/helm_sync_test.go:106]: Ah I didn't realize `DefaultHelmReleaseNamespace` was defined as `default`. Ack, thank you for explaining

**Comment by sdowell** [comment]
LGTM, but I'll give @karlkfi a chance to finish his review

File: e2e/testcases/helm_sync_test.go
Comment (nan-yu) [reply e2e/testcases/helm_sync_test.go:341]: If user's workload creates new K8s resources and doesn't have a finalizer to remove them, then it is a problem.
The PVC seems to be created by MariaDB, which is not a field specified in the Deployment.
Let me check the chart values to see whether we can disable persistence.

**Comment by nan-yu** [comment]
/hold

Exploring the option to disable persistence.

File: e2e/testcases/helm_sync_test.go
Comment (nan-yu) [reply e2e/testcases/helm_sync_test.go]: Done. Updated the chart value to disable mariadb persistent storage.

Comment (karlkfi) [reply e2e/testcases/helm_sync_test.go]: New config looks good!

File: e2e/testdata/root-sync-helm-chart-cr.yaml
Comment (sdowell) [inline e2e/testdata/root-sync-helm-chart-cr.yaml:48]: Nice! Will we still need to manually clean up the lingering `PVC` resources on the clusters from previous runs?

Comment (nan-yu) [reply e2e/testdata/root-sync-helm-chart-cr.yaml:48]: The cleanup function is no longer needed and removed.

Comment (sdowell) [reply e2e/testdata/root-sync-helm-chart-cr.yaml:48]: I mean specifically for the resources that were created before this change, i.e. test runs on commits before this change is submitted. Will we need to manually clean up those?

**Review by karlkfi — APPROVED** [review]
/lgtm

File: e2e/testdata/root-sync-helm-chart-cr.yaml
Comment (nan-yu) [reply e2e/testdata/root-sync-helm-chart-cr.yaml:48]: That should not be needed as the mariadb doesn't check the persisted data, but it is good to delete the test data anyway.
I'll remove them later once the first CI job passes after this PR is merged.

**Pull request merged**
